File: trunk/ShadowFrameworkLite20/src/shadow/system/data/objects/SFVectorData.java
package shadow.system.data.objects;

import shadow.system.data.SFInputStream;
import shadow.system.data.SFOutputStream;

public class SFVectorData extends SFPrimitiveType{

	private float[] floatValues;
	
	protected SFVectorData(int floatValuesSize) {
		this.floatValues = new float[floatValuesSize];
	}

	public float[] getFloatValues() {
		return floatValues;
	}

	@Override
	public void readFromStream(SFInputStream stream) {
		float[] values=stream.readFloats(floatValues.length);
		for (int i = 0; i < values.length; i++) {
			floatValues[i]=values[i];
		}
	}
	
	@Override
	public void writeOnStream(SFOutputStream stream) {
		stream.writeFloats(floatValues);
	}

	@Override
	public SFVectorData clone(){
		return new SFVectorData(floatValues.length);
	}
}

File: trunk/ShadowFrameworkJS/src/shadow/system/data/objects/SFVectorData.js
//Java to JS on 15/03/2012

function SFVectorData(){
	this.floatValues=new Array();
}

inherit(SFVectorData,SFPrimitiveType);


SFVectorData.prototype["getFloatValues"]=function(){
			return floatValues;
		};
		
SFVectorData.prototype["readFromStream"]=function(stream){
			var values = stream.readFloats(this.floatValues.length);
			for ( var  i  =  0 ; i   < values.length ; i++ ){
				this.floatValues[i]  = values[i];
			}
		};
		
SFVectorData.prototype["writeOnStream"]=function(stream){
			stream.writeFloats(floatValues);
		};

SFVectorData.prototype["clone"]=function(){
			return new SFVectorData();
		};
